.github/workflows/deploy_release.yml

```diff
@@ -0,0 +1,77 @@
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more
+#  contributor license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright ownership.
+#  The ASF licenses this file to You under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with
+#  the License. You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+name: Deploy Release
+
+# This workflow is intended to be run after successfully validation of the release
+# staged by stage_release.  It will release (or drop) the staging repository.
+#
+#
+# It requires the following:
+# variables:
+# KROXYLICIOUS_SONAR_TYPE_TOKEN_USERNAME - Sonar Type Access User Token Username
+# secrets:
+# KROXYLICIOUS_SONAR_TYPE_TOKEN_PASSWORD - Sonar Type Access User Token Password
+# KROXYLICIOUS_RELEASE_TOKEN             - GitHub PAT wih content/createPullRequest permission for Kroxylicious repo.
+#
+
+on:
+  workflow_dispatch:
+    inputs:
+      next-state:
+        description: 'Desired next state for the staging repository'
+        required: true
+        type: choice
+        default: drop
+        options:
+          - drop
+          - release
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'Check out repository'
+        uses: actions/checkout@v4
+        with:
+          ref: main # transition-staging-repository-state.sh requires only the pom.xml, being on the release branch is not important
+          token: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          server-id: ossrh
+          server-username: SONAR_TYPE_TOKEN_USERNAME # env variable for Sonar Type username
```

Comment:
Maven central is hosted by `Sonatype` not `Sonar Type`. Sona**r** is a different business entirely. 

I was confused why we needed credentials for SonarCloud here for minute.

Reply:
I had the two business conflated in my mind.